force-app/main/default/objects/Public_Event__c/fields/Street__c.field-meta.xml

```diff
@@ -1,9 +1,9 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <CustomField xmlns="http://soap.sforce.com/2006/04/metadata">
-    <fullName>Address__c</fullName>
```

Comment:
There should be separate metadata for address field. Otherwise it will not be deleted from the target org.